src/script/conversation/ConversationRepository.ts

```diff
@@ -872,17 +872,27 @@ export class ConversationRepository {
    * Check for conversation locally and fetch it from the server otherwise.
    * TODO(Federation): Remove "optional" from "domain"
    */
-  async getConversationById(conversation_id: QualifiedId): Promise<Conversation> {
+  async getConversationById(conversation_id: QualifiedId, searchInLocalDB = false): Promise<Conversation> {
     if (typeof conversation_id.id !== 'string') {
       throw new ConversationError(
         ConversationError.TYPE.NO_CONVERSATION_ID,
         ConversationError.MESSAGE.NO_CONVERSATION_ID,
       );
     }
-    const conversationEntity = this.conversationState.findConversation(conversation_id);
-    if (conversationEntity) {
-      return conversationEntity;
+    const localStateConversation = this.conversationState.findConversation(conversation_id);
+    if (localStateConversation) {
+      return localStateConversation;
+    }
+
+    if (searchInLocalDB) {
+      const localDBConversation = await this.conversationService.loadConversation<BackendConversation>(
+        conversation_id.id,
+      );
+      if (localDBConversation) {
+        return this.mapConversations([localDBConversation])[0];
+      }
```

Comment:
If the conversation was filtered out and not included in the local state, we try to load it directly from indexedDB, so the incoming event regarding it can be executed.